torch/testing/_internal/custom_op_db.py

```diff
@@ -0,0 +1,126 @@
+import torch
+from torch.testing._internal.opinfo.core import (
+    OpInfo,
+)
+from torch.testing._internal.common_dtype import all_types_and
+import numpy as np
+from torch._custom_op import CustomOp
+from torch.testing._internal.autograd_function_db import (
+    sample_inputs_numpy_cube,
+    sample_inputs_numpy_mul,
+    sample_inputs_numpy_sort,
+    sample_inputs_numpy_take,
+)
+
+# Note: [custom op db]
+#
+# This is a collection of custom operator test cases written as OpInfos
+# so they can easily be consumed by OpInfo-based tests to check if subsystems
+# support them correctly.
+
+def to_numpy(tensor):
+    return tensor.cpu().numpy()
+
+@CustomOp.define('(Tensor x) -> (Tensor, Tensor)', ns='_torch_testing')
+def numpy_cube(x):
+    ...
+
+@numpy_cube.impl('cpu')
+@numpy_cube.impl('cuda')
+def numpy_cube_impl(x):
+    x_np = to_numpy(x)
+    dx = torch.tensor(3 * x_np ** 2, device=x.device)
+    return torch.tensor(x_np ** 3, device=x.device), dx
+
+@numpy_cube.impl_meta()
+def numpy_cube_meta(x):
+    return x.clone(), x.clone()
+
+@CustomOp.define('(Tensor x, Tensor y) -> Tensor', ns='_torch_testing')
```

Comment:
minor API nit: `@custom_op(schema)` seems more direct, and the user doesn't really care about the CustomOp class.

Reply:
Or maybe, you should create a namespace, and then use that to define operators on the namespace.

Reply:
I agree that `@custom_op(schema)` is more direct, but in the context of the three entrypoints (only the first exists right now), it makes sense that they are accessed similarly:
- Entrypoint 1: decorator CustomOp.define (for your new custom ops)
- Entrypoint 2: decorator CustomOp.from_decomp (for your custom operators that are composed of only PyTorch operations, we can autogenerate rules for them)
- Entrypoint 3: function CustomOp.from_existing (pass in an operator registered with TORCH_LIBRARY/torch.library here)

Maybe we can combine entrypoint 1 and 2 into a `@custom_op(schema, *, ns, decomp=False)` and actually use the function the user provided to us as the decomp when implementations don't exist and come up with a succinct name for Entrypoint 3? ("as_custom_op"?)

Reply:
yeah I would just add kwargs to `custom_op`, or just have another function entirely

For preexisting function, you actually probably want to hang it on torch.ops, TBH. Because you don't actually want the custom_op outer function.